virttest/migration.py

```diff
@@ -578,6 +579,29 @@ def set_migratepostcopy(self, vm_name, uri=None):
             raise exceptions.TestError("Unalbe to find event {}"
                                        .format(exp_str))
 
+    def control_migrate_speed(self, vm_name, to_speed=1, opts=""):
+        """
+        Set migration speed to control the migration duration
+
+        :param vm_name: vm name
+        :param to_speed: the speed value in Mbps to be set for migration
+        :param opts: the options for virsh command
+        """
+
+        def _set_speed(vm_name, to_speed, opts=''):
+            virsh_args = {}
+            virsh_args.update({"ignore_status": False})
+            old_speed = virsh.migrate_getspeed(vm_name, extra=opts, **virsh_args)
+            logging.debug("Current %s migration speed is %s MiB/s\n", opts, old_speed.stdout_text.strip())
+            logging.debug("Set %s migration speed to %d MiB/s\n", opts, to_speed)
+            virsh.migrate_setspeed(vm_name, to_speed, extra=opts, **virsh_args)
+            actual_speed = virsh.migrate_getspeed(vm_name, extra=opts, **virsh_args)
```

Comment:
Does it necessary to check if the speed is set as expected?

Reply:
updated